grails-app/services/org/pih/warehouse/replenishment/ReplenishmentService.groovy

```diff
@@ -37,10 +37,26 @@ class ReplenishmentService {
     boolean transactional = true
 
     def getRequirements(Location location, InventoryLevelStatus inventoryLevelStatus) {
-        return Requirement.createCriteria().list() {
+        List<Requirement> requirements = []
+        Requirement.createCriteria().list() {
             eq("location", location)
             eq("status", inventoryLevelStatus)
+        }.collect { Requirement requirement ->
```

Comment:
why are you using collect here instead of findAll?

Reply:
I thought it will be an easier and cleaner way of getting rid of duplicates.

Reply:
in this case unique() should be fine

Reply:
Agree with @pmuchowski. Seems like each (iterate) would be better than collect (transform), but unique seems like the better option.

Reply:
fixed